runtime/json-writer.h

```diff
@@ -0,0 +1,67 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2020 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#pragma once
+
+#include "runtime/kphp_core.h"
+
+#include <array>
+#include <string_view>
+
+namespace impl_ {
+
+class JsonWriter : vk::not_copyable {
+public:
+  explicit JsonWriter(bool pretty_print = false) noexcept;
+  ~JsonWriter() noexcept;
+
+  bool Bool(bool b) noexcept;
+  bool Int64(int64_t i) noexcept;
+  bool Double(double d) noexcept;
+  bool String(const string &s) noexcept;
+  bool RawString(const string &s) noexcept;
+  bool Null() noexcept;
+
+  bool Key(std::string_view key) noexcept;
+  bool StartObject() noexcept;
+  bool EndObject() noexcept;
+  bool StartArray() noexcept;
+  bool EndArray() noexcept;
+
+  void SetDoublePrecision(std::size_t precision) noexcept {
+    double_precision_ = precision;
+  }
+
+  bool IsComplete() const noexcept;
+  string GetError() const noexcept;
+  string GetJson() const noexcept;
+
+private:
+  bool newLevel(bool is_array) noexcept;
+  bool exitLevel(bool is_array) noexcept;
+  bool RegisterValue() noexcept;
+  bool SetString(const string &s, bool with_quotes) noexcept;
```

Comment:
Why do you name some methods camelCase(), but some CapitalCase()?

Reply:
fixed